modules/third-party/org-freemarker/patches/LPS-160230.patch

```diff
@@ -0,0 +1,49 @@
+diff --git a/freemarker/ext/jsp/JspTagModelBase.java b/freemarker/ext/jsp/JspTagModelBase.java
+index 032cead0ab0e..eed58a07e9f5 100644
+--- a/freemarker/ext/jsp/JspTagModelBase.java
++++ b/freemarker/ext/jsp/JspTagModelBase.java
+@@ -19,10 +19,6 @@
+ 
+ package freemarker.ext.jsp;
+ 
+-import java.beans.BeanInfo;
+-import java.beans.IntrospectionException;
+-import java.beans.Introspector;
+-import java.beans.PropertyDescriptor;
+ import java.lang.reflect.InvocationTargetException;
+ import java.lang.reflect.Method;
+ import java.math.BigDecimal;
+@@ -50,16 +46,19 @@ class JspTagModelBase {
+     private final Method dynaSetter;
+     private final Map propertySetters = new HashMap();
+     
+-    protected JspTagModelBase(String tagName, Class tagClass) throws IntrospectionException {
++    protected JspTagModelBase(String tagName, Class tagClass) {
+         this.tagName = tagName;
+         this.tagClass = tagClass;
+-        BeanInfo bi = Introspector.getBeanInfo(tagClass);
+-        PropertyDescriptor[] pda = bi.getPropertyDescriptors();
+-        for (int i = 0; i < pda.length; i++) {
+-            PropertyDescriptor pd = pda[i];
+-            Method m = pd.getWriteMethod();
+-            if (m != null) {
+-                propertySetters.put(pd.getName(), m);
++        for (Method method : tagClass.getMethods()) {
++            if (method.getReturnType().equals(void.class) &&
++                (method.getParameterTypes().length == 1) &&
++                method.getName().matches("^set[A-Z].*")) {
```

Comment:
@guilhermedcamacho please get rid of this regex usage, just do plain String startWith and length testing. Also please adjust the commit order. The patching commit should go right after the initial checkin.